realm/realm-library/src/main/java/io/realm/internal/TableQuery.java

```diff
@@ -703,12 +705,59 @@ public long count() {
         return nativeCount(nativePtr);
     }
 
-    public void rawPredicate(String filter, @Nullable OsKeyPathMapping mapping, DescriptorOrdering descriptors, String[] args) {
+    public void rawPredicate(String filter, @Nullable OsKeyPathMapping mapping, DescriptorOrdering descriptors, Object[] args) {
+        long listPtr = nativeCreateArgumentList();
```

Comment:
I was hoping we could have used the Java Mixed class so we could just do something like `long[] args` and then read the underlying mixed value? It would be slightly slower as we needed to create the mixed,  but this implementation would be trivial.